### PR TITLE
Fixed `linkerd check` not finding Prometheus

### PR DESCRIPTION
**Pull request by alpeb**
## The Problem

`linkerd check` run right after install is failing because it can't find the Prometheus Pod.

## The Cause

The "control plane pods are ready" check used to verify the existence of all the control plane pods, blocking until all the pods were ready.

Since #4724, Prometheus is no longer included in that check because it's checked separately as an add-on. An unintended consequence is that when the ensuing "control plane self-check" is triggered, Prometheus might not be ready yet and the check fails because it doesn't do retries.

## The Fix

The "control plane self-check" uses a gRPC call (it's the only check that does that) and those weren't designed with retries in mind.

This PR adds retry functionality to the `runCheckRPC()` function, making sure the final output remains the same

It also temporarily disables the `upgrade-edge` integration test because after installing edge-20.7.4 `linkerd check` will fail because of this.

### CODE REVIEW

File: pkg/healthcheck/healthcheck.go [hunk 1]
```diff
@@ -1038,10 +1038,11 @@ func (hc *HealthChecker) allCategories() []category {
 					},
 				},
 				{
-					description:   "control plane self-check",
-					hintAnchor:    "l5d-api-control-api",
-					fatal:         true,
-					retryDeadline: hc.RetryDeadline,
+					description:         "control plane self-check",
+					hintAnchor:          "l5d-api-control-api",
+					surfaceErrorOnRetry: false,
```
Comment (Pothulapati) [inline pkg/healthcheck/healthcheck.go:1043]: Any reason this has to be false? As its hard to understand which check we are waiting for with just `\ waiting for check to complete`?

Comment (Pothulapati) [reply pkg/healthcheck/healthcheck.go:1043]: I get the following output, by making it `true`
```
- Error calling Prometheus from the control plane: server_error: server error: 503
```
This seems better than the waiting for check right?

File: pkg/healthcheck/healthcheck.go [hunk 2]
```diff
@@ -1353,49 +1354,75 @@ func (hc *HealthChecker) runCheck(categoryID CategoryID, c *checker, observer Ch
...
+				checkResult.Retry = time.Now().Before(c.retryDeadline)
+				// only show the waiting message during retries,
+				// and send the underlying error on the last try
+				if !c.surfaceErrorOnRetry && checkResult.Retry {
+					checkResult.Err = errors.New("waiting for check to complete")
+				}
+				observer(checkResult)
+			} else if !checkResult.alreadyObserved(observedResults) {
+				observer(checkResult)
+			}
```
[the remark this answers is not in the document]
Comment (alpeb) [reply pkg/healthcheck/healthcheck.go:1414]: The control plane self-check (which is the only check that is using `checkRPC`) returns two responses, one for the connectivity check against k8s and another for the connectivity check against prometheus. This inner `for` might have already run in a previous try (outer `for`) with a success against k8s and a failure against prom, and we don't wanna report the success again. Without this check `linkerd check` would show something like:
```
linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
√ [kubernetes] control plane can talk to Kubernetes
√ [kubernetes] control plane can talk to Kubernetes
√ [kubernetes] control plane can talk to Kubernetes
√ [kubernetes] control plane can talk to Kubernetes
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus
√ tap api service is running
```

File: pkg/healthcheck/healthcheck.go (hunk 1 above)
Comment (alpeb) [reply pkg/healthcheck/healthcheck.go:1043]: As you know, prometheus takes a long while to become available, so users that run `linker check` right after install will always see that error. I thought it was less scary for them to just let them know we're waiting for the check. If prometheus doesn't come up by the time-out, the real error is shown:
```
linkerd-api
-----------
√ control plane pods are ready
√ control plane self-check
√ [kubernetes] control plane can talk to Kubernetes
× [prometheus] control plane can talk to Prometheus
    Error calling Prometheus from the control plane: server_error: server error: 503
    see https://linkerd.io/checks/#l5d-api-control-api for hints

Status check results are ×
```

File: pkg/healthcheck/healthcheck.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kleimkuhler) [reply pkg/healthcheck/healthcheck.go:1414]: Ah cool thanks for the explanation!
